desk_pin_table.py

```diff
@@ -72,12 +71,24 @@ def register_pin_events(self):
                 # TODO: Add some type of custom debouncing functionality, or use bouncetime = 200 below
                 GPIO.add_event_detect(pin, GPIO.BOTH, lambda channel, desk_obj=desk_obj: desk_obj.input_received(channel, GPIO.input(channel)), bouncetime=50)
 
-        return None
-
-    '''
-    Use Desk mapping to lookup a desk object given a desk ID
-    '''
-    def get_desk_from_id(self, id):
-        for (desk_id, pin, desk_obj) in self.__desk_mapping:
-            if id == desk_id:
-                return desk_obj
\ No newline at end of file
+    def get_desk_from_id(self, desk_id):
+        """
+        Lookup a Desk Object given it's ID
+        :param desk_id: The Desk ID to look for
+        :return: The Desk object if the ID corresponds to a Desk, or None
+        """
+        t = filter(lambda info: desk_id == info[0], self.__desk_mapping)
+
+        for (d_id, pin, d_obj) in t:
+            return d_obj
```

Comment:
I looked up [how to get the first element in a list that matches a condition](https://stackoverflow.com/questions/2361426/get-the-first-item-from-an-iterable-that-matches-a-condition)

Turns out the solution is
`next(d_obj for d_id, pin, d_obj in self.__desk_mapping if d_id == desk_id)`